src/main/scala/edu/ie3/simona/service/ev/ExtEvDataService.scala

```diff
@@ -298,8 +301,10 @@ class ExtEvDataService(override val scheduler: ActorRef)
   ): (ExtEvStateData, Option[Long]) = {
 
     if (tick == INIT_SIM_TICK) {
+      // During initialization, an empty ProvideArrivingEvs message
+      // is sent, which includes the first relevant tick
 
-      maybeNextTick.getOrElse(
+      val nextTick = maybeNextTick.getOrElse(
```

Comment:
this is the reason because maybeNextTick needs to be Option[Long]? Else maybe there is a possibility to directly provide a nextTick and get rid of this getOrElse?

Reply:
During initialization, MobSim has to provide a (first) next tick, otherwise the simulation can't start. This is why we require the Optional to be non-empty and we extract the tick here. After the first simulation tick, MobSim is allowed to not provide a next tick, which signifies the end of the simulation.

Reply:
Alright, thanks for the explanation